csharp/src/Microsoft.ML.OnnxRuntime/InferenceSession.cs

```diff
@@ -114,98 +114,561 @@ public IDisposableReadOnlyCollection<DisposableNamedOnnxValue> Run(IReadOnlyColl
+            IReadOnlyCollection<FixedBufferOnnxValue> inputValues,
+            IReadOnlyCollection<string> outputNames,
+            IReadOnlyCollection<FixedBufferOnnxValue> outputValues,
+            RunOptions options)
+        {
+            if (inputNames.Count != inputValues.Count)
+            {
+                throw new ArgumentException($"Length of {nameof(inputNames)} ({inputNames.Count}) must match that of {nameof(inputValues)} ({inputValues.Count}).");
+            }
+            if (outputNames.Count != outputValues.Count)
+            {
+                throw new ArgumentException($"Length of {nameof(outputNames)} ({outputNames.Count}) must match that of {nameof(outputValues)} ({outputValues.Count}).");
+            }
+
+            // prepare inputs
+            string[] inputNamesArray = inputNames as string[] ?? inputNames.ToArray();
+            IntPtr[] inputValuesArray = new IntPtr[inputNames.Count];
+            int inputIndex = 0;
+            foreach (var input in inputValues)
+            {
+                inputValuesArray[inputIndex] = input.Value;
+
+                inputIndex++;
+            }
+
+            // prepare outputs
+            string[] outputNamesArray = outputNames as string[] ?? outputNames.ToArray();
+            IntPtr[] outputValuesArray = new IntPtr[outputNames.Count];
+            int outputIndex = 0;
+            foreach (var output in outputValues)
+            {
+                outputValuesArray[outputIndex] = output.Value;
+
+                outputIndex++;
+            }
+
+            IntPtr status = NativeMethods.OrtRun(
+                                                _nativeHandle,
+                                                options.Handle,
+                                                inputNamesArray,
+                                                inputValuesArray,
+                                                (UIntPtr)inputNames.Count,
+                                                outputNamesArray,
+                                                (UIntPtr)outputNames.Count,
+                                                outputValuesArray /* pointers to Pre-allocated OrtValue instances */
+                                                );
+
+            NativeApiStatus.VerifySuccess(status);
+        }
+
+        /// <summary>
+        /// Runs the loaded model for the given inputs and outputs.
+        /// 
+        /// Outputs need to be created with correct type and dimension to receive the fetched data.
+        /// </summary>
+        /// <param name="inputs">Specify a collection of <see cref="NamedOnnxValue"/> that indicates the input values.</param>
+        /// <param name="output">Specify a collection of <see cref="NamedOnnxValue"/> that indicates the output values.</param>
+        public void Run(
+            IReadOnlyCollection<NamedOnnxValue> inputs,
+            IReadOnlyCollection<NamedOnnxValue> outputs)
```

Comment:
So the only benefit of using this override of Run() when compared with the one that produces FixedBufferOnnxValue is that the user doesn't need to dispose the instance and the underlying OrtValue is destroyed automatically in this case ?